service/controller/internal/vmsku/vmsku.go

```diff
@@ -0,0 +1,107 @@
+package vmsku
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/azure-operator/v4/client"
+)
+
+const (
+	// CapabilitySupported is the value returned by this API from Azure when the capability is supported
+	CapabilitySupported = "True"
+
+	CapabilityAcceleratedNetworking = "AcceleratedNetworkingEnabled"
+)
+
+type Config struct {
+	ClientFactory *client.Factory
+	Location      string
+	Logger        micrologger.Logger
+}
+
+type Interface struct {
+	clientFactory *client.Factory
+	location      string
+	skus          map[string]*compute.ResourceSku
+	logger        micrologger.Logger
+}
+
+func New(config Config) (*Interface, error) {
+	if config.ClientFactory == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.ClientFactory must not be empty", config)
+	}
+	if config.Location == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Location must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	return &Interface{
+		clientFactory: config.ClientFactory,
+		location:      config.Location,
+		logger:        config.Logger,
+	}, nil
+}
+
+func (v *Interface) HasCapability(ctx context.Context, vmType string, name string) (bool, error) {
+	if len(v.skus) == 0 {
+		err := v.initCache(ctx)
+		if err != nil {
+			return false, microerror.Mask(err)
+		}
+	}
```

Comment:
Would it make sense to wrap this in a function like `v.ensureInititialized()` which would then do `v.initCache()` call once inside an instance specific mutex to make it read nicer and prevent concurrent init calls?

Reply:
done